eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Pool.hs

```diff
@@ -293,9 +294,14 @@ poolDelegationTransition = do
           -- has been removed from the registered pools).  does it need to pay a
           -- new deposit (at the current deposit amount). But of course, if that
           -- has happened, we cannot be in this branch of the case statement.
+          let futureStakePoolState =
+                mkStakePoolState
+                  (stakePoolState ^. spsDepositL)
+                  (stakePoolState ^. spsDelegsL)
```

Comment:
This is actually incorrect, because we need to use the value of that delegators set not from when certificate was submitted, but when the actual rotation of future to current happens. This is because by the time future becomes current new delagation might have been added and old ones removed! In other words we need to update this in the POOLREAP instead: https://github.com/IntersectMBO/cardano-ledger/blob/f776625c3ddf1bd0a3873d1683265b97e6733953/eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/PoolReap.hs#L155
```haskell
psStakePools = 
  Map.unionWith
   (\newPoolState oldPoolState -> newPoolState {spsDelegators = spsDelegators oldPoolState)
   (ps0 ^. psFutureStakePoolsL)
   (ps0 ^. psStakePoolsL)
```

```suggestion
                  mempty
```